src/ado2gh/Commands/GenerateScriptCommand.cs

```diff
@@ -171,36 +177,61 @@ public async Task Invoke(string githubOrg, string adoOrg, FileInfo output, bool
             return pipelines;
         }
 
-        public async Task<IDictionary<string, IDictionary<string, IEnumerable<string>>>> GetRepos(AdoApi ado, IEnumerable<string> orgs)
+        public async Task<IDictionary<string, IDictionary<string, IEnumerable<string>>>> GetRepos(AdoApi ado, IEnumerable<string> orgs, string adoTeamProject)
         {
             var repos = new Dictionary<string, IDictionary<string, IEnumerable<string>>>();
 
             if (orgs != null && ado != null)
             {
+                var teamProjectExists = false;
                 foreach (var org in orgs)
                 {
                     _log.LogInformation($"ADO ORG: {org}");
                     repos.Add(org, new Dictionary<string, IEnumerable<string>>());
 
                     var teamProjects = await ado.GetTeamProjects(org);
-
-                    foreach (var teamProject in teamProjects)
+                    if (string.IsNullOrEmpty(adoTeamProject))
                     {
-                        _log.LogInformation($"  Team Project: {teamProject}");
-                        var projectRepos = await ado.GetEnabledRepos(org, teamProject);
-                        repos[org].Add(teamProject, projectRepos);
-
-                        foreach (var repo in projectRepos)
+                        foreach (var teamProject in teamProjects)
+                        {
+                            _log.LogInformation($"  Team Project: {teamProject}");
+                            teamProjectExists = true;
+                            var projectRepos = await GetTeamProjectRepos(ado, org, teamProject);
+                            repos[org].Add(teamProject, projectRepos);
+                        }
+                    }
+                    else
+                    {
+                        if (teamProjects.Any(o => o.Equals(adoTeamProject, StringComparison.OrdinalIgnoreCase)))
                         {
-                            _log.LogInformation($"    Repo: {repo}");
+                            _log.LogInformation($"  Team Project: {adoTeamProject}");
+                            teamProjectExists = true;
+                            var projectRepos = await GetTeamProjectRepos(ado, org, adoTeamProject);
+                            repos[org].Add(adoTeamProject, projectRepos);
                         }
                     }
                 }
+                if (!teamProjectExists)
+                {
+                    _log.LogWarning($"ADO Team Project provided cannot be found [{adoTeamProject}]");
+                }
             }
 
             return repos;
         }
 
+        private async Task<IEnumerable<string>> GetTeamProjectRepos(AdoApi ado, string org, string teamProject)
+        {
+            _log.LogInformation($"  Team Project: {teamProject}");
```

Comment:
I think this is going to result in redundant log output with line 207

Reply:
updated, thanks, was the same on line 197 🙂